src/fights.ts

```diff
@@ -1387,25 +1390,41 @@ const freeFightSources = [
   new FreeFight(
     () => {
       if (!have($item`closed-circuit pay phone`)) return false;
+      // Check if we have or can get Shadow Affinity
       if (have($effect`Shadow Affinity`)) return true;
-      if (get("_shadowAffinityToday")) return false;
+      if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget()) return true;
 
-      if (!ClosedCircuitPayphone.rufusTarget()) return true;
       if (get("rufusQuestType") === "items") {
         return false; // We deemed it unprofitable to complete the quest in potionSetup
       }
       if (get("encountersUntilSRChoice") === 0) {
         // Target is either an artifact or a boss
         return true; // Get the artifact or kill the boss immediately for free
       }
-      return false; // We have to spend turns to get the artifact or kill the boss
+
+      // Consider forcing noncombats below:
+      if (globalOptions.prefs.yachtzeechain) return false; // NCs are better when yachtzeeing, probably
+      // TODO: With the KoL update, is there a function for checking if an NC is already forced?
```

Comment:
this doesn't exist yet but in theory should exist soon, i guess? 

i don't think this needs to happen for initial adoption of this PR but [the way soolar parses the charpane for NC forces](https://github.com/Loathing-Associates-Scripting-Society/ChIT/blob/424aba089e87b7e66b4762d97a1fb2cd0246df0f/src/relay/charpane.ash#L1202) is a good example for how libram or garbo might do the same.

Reply:
it's in api.php now